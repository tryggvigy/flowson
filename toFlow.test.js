const {toFlow, setTypeName} = require('./toFlow');

describe('toFlow', () => {
  it('handles empty objects', () => {
    expect(toFlow({})).toEqual('{}')
  })
  
  it('transforms json as expected', () => {
    expect(toFlow({
      str: 'cake',
      number: 1,
      bool: false,
      arr: [1, true, 'foo', {}, [1, {foo: 'bar'}]],
      nestedObj: {
        arr: [1, true, 'foo', {}, [1, {foo: 'bar'}]]
      }
    })).toMatchSnapshot()
  })
})

describe('setTypeName', () => {
  it('defaults to the filename passed, with stripped extension and postfixed `T`', () => {
    expect(
      setTypeName('{"name": string}', 'config.json', undefined)
    ).toMatchSnapshot()
  })
  
  it('uses the user defined type name if passed', () => {
    expect(
      setTypeName('{"name": string}', 'config.json', 'stupidName')
    ).toMatchSnapshot()
  })
})