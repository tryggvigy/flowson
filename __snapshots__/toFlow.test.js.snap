// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setTypeName defaults to the filename passed, with stripped extension and postfixed \`T\` 1`] = `
"// @flow
export type configT = {\\"name\\": string}"
`;

exports[`setTypeName uses the user defined type name if passed 1`] = `
"// @flow
export type stupidName = {\\"name\\": string}"
`;

exports[`toFlow transforms json as expected 1`] = `
"{
  \\"name\\": string,
}"
`;
